### PR TITLE
Alibaba OSS support in Multi-Cloud Gelato Project

### PR DESCRIPTION
**Pull request by Click2Cloud-Gamma**
In Gelato project of SODA foundation, currently it supports to perform object operations, lifecycle management and object migration on various cloud vendors like AWS, Azure, GCP, IBM, Huawei, YIG and Ceph. So adding Alibaba as a backend will allow user to perform object management, object lifecycle management (in-cloud & cross-cloud) and migration for Alibaba Cloud using Gelato(multi-cloud) Project for Alibaba OSS Service.

This is  design-spec for details of features that Alibaba OSS will support as new backend in Gelato Project

### CODE REVIEW

**Comment by skdwriting** [comment]
@Click2Cloud-Gamma 
Why there is no description to the PR?
How about the previous PR #51 comments are closed? no comment updates

File: specs/daito/Alibaba_OSS_Support.md [hunk 1]
```diff
@@ -0,0 +1,329 @@
...
+* SSE supported
+* Upload Object on Alibaba OSS (PUT Object)  
+* Download Object from Alibaba OSS (GET Object)
+* Support for Multi-part Upload
+* Delete Object from Alibaba OSS
+* Lifecycle management by Alibaba OSS (Standard, IA, Archive)
+
+| | Tier_1 | Tier_99 | Tier_999 |
+|---------------|---------------|---------------|---------------|
+| Alibaba OSS | STANDARD | IA | Archive |
```
Comment (skdwriting) [inline specs/daito/Alibaba_OSS_Support.md:29]: please provide the details on the storage classes and attributes, compared to exsiting?

File: specs/daito/Alibaba_OSS_Support.md [hunk 2]
```diff
@@ -0,0 +1,329 @@
...
+![picture](Alibaba_backend.png)
+
+                    Figure: Architecture of Integration of Alibaba Cloud in OpenSDS/Multi-cloud
+
+By adding Alibaba OSS as adapter  in OpenSDS will help users to perform all bucket and object related operations as follow:
+* Add Backend for Alibaba OSS (`alibaba-oss` type)
+* Create OpenSDS Bucket on Alibaba OSS
+* Delete Bucket from Alibaba OSS
+* SSE supported
+* Upload Object on Alibaba OSS (PUT Object)
```
Comment (skdwriting) [inline specs/daito/Alibaba_OSS_Support.md:21]: This part is analysed? still issues there?

**Comment by Click2Cloud-Gamma** [comment]
> @Click2Cloud-Gamma
> Why there is no description to the PR?
> How about the previous PR #51 comments are closed? no comment updates

@skdwriting  Yes, I have added all the requested changes in this design-spec document. Thanks

File: specs/daito/Alibaba_OSS_Support.md (hunk 2 above)
Comment (Click2Cloud-Gamma) [reply specs/daito/Alibaba_OSS_Support.md:21]: No we have tested with latest Elba release too, it works fine. Thanks

File: specs/daito/Alibaba_OSS_Support.md (hunk 1 above)
Comment (Click2Cloud-Gamma) [reply specs/daito/Alibaba_OSS_Support.md]: Added comparison with storage classes of other supported cloud. Thanks

**Comment by skdwriting** [comment]
The previous review comments are closed. Unable to see it from here as the earlier PR is closed.. Could you please check with reviewers?

**Comment by Click2Cloud-Gamma** [comment]
> The previous review comments are closed. Unable to see it from here as the earlier PR is closed.. Could you please check with reviewers?

FIxed all the requested changes and moved files to `specs` folder as requested. Please review, Thanks

**Pull request merged**